lib/suppressions.cpp

```diff
@@ -448,19 +448,19 @@ const std::list<Suppressions::Suppression> &Suppressions::getSuppressions() cons
 }
 
 /**
- * This function require that both tokens and suppressions are sorted in line
- * number order.
+ * Loop on mSuppressions over and oer again might not be so fast.
  */
 void Suppressions::markUnmatchedInlineSuppressionsAsChecked(const Tokenizer &tokenizer) {
-    int lastLineNr = 0;
-    auto supp = mSuppressions.begin();
-    for (const Token *tok = tokenizer.tokens(); tok && (supp != mSuppressions.end()); tok = tok->next()) {
-        if (tok->linenr() != lastLineNr) {
-            lastLineNr = tok->linenr();
-            while ((supp != mSuppressions.end()) && (supp->lineNumber <= lastLineNr)) {
-                if (supp->lineNumber == lastLineNr)
-                    supp->checked = true;
-                ++supp;
+    int currLineNr = -1;
+    int currFileIdx = -1;
+    for (const Token *tok = tokenizer.tokens(); tok; tok = tok->next()) {
+        if (currFileIdx != tok->fileIndex() || currLineNr != tok->linenr()) {
+            currLineNr = tok->linenr();
+            currFileIdx = tok->fileIndex();
+            for (auto &supp : mSuppressions) {
+                if (supp.fileName == tokenizer.list.file(tok) && supp.lineNumber == currLineNr) {
```

Comment:
Maybe you can put the lineNumber comparison before the filename comparison. A integer comparison should be faster than a string comparison.

Reply:
It is run one time per configuration for every file. Even if hash would skip other checks. I put it before that check cause I had problem with some suppression's not getting their check flag set. But that is mostly because putting 'same' test in different ifdef combination. In the 'real' world that maybe not happening so often.

Also, maybe some debug flag should print suppression's that does not have the checked flag set.